sys/include/can/can.h

```diff
@@ -95,12 +95,36 @@ struct can_frame {
     uint8_t data[CAN_MAX_DLEN] __attribute__((aligned(8)));
 };
 
+#if defined(MCU_SAMD5X)
+typedef enum {
+    CAN_FILTER_TYPE_RANGE = 0x00,   /**< Range filter from Filter 1 to Filter 2 (Filter 2 > Filter 1) */
+    CAN_FILTER_TYPE_DUAL,           /**< Dual ID Filter (Filter 2 or Filter 1) */
+    CAN_FILTER_TYPE_CLASSIC,        /**< Classic Filter: Filter ID and Mask */
+    CAN_FILTER_TYPE_EXT_RANGE       /**< For extended filters, Range filter from Filter 1 to Filter 2 (Filter 2 > Filter 1) */
+} can_filte_type_t;
+
+typedef enum {
+    CAN_FILTER_DISABLE = 0x00,      /**< Disable Filter element */
+    CAN_FILTER_RX_FIFO_0,           /**< Store message in Rx FIFO 0 if filter matches */
+    CAN_FILTER_RX_FIFO_1,           /**< Store message in Rx FIFO 1 if filter matches */
+    CAN_FILTER_RX_REJECT,           /**< Reject message if filter matches */
+    CAN_FILTER_RX_PRIO,             /**< Set priority if filter matches */
+    CAN_FILTER_RX_PRIO_FIFO_0,      /**< Set priority and store message in Rx FIFO 0 if filter matches */
+    CAN_FILTER_RX_PRIO_FIFO_1,      /**< Set priority and store message in Rx FIFO 1 if filter matches */
+    CAN_FILTER_RX_STRXBUF           /**< Store message in the RX buffer or as debug message */
+} can_filter_conf_t;
+#endif
+
 /**
  * @brief Controller Area Network filter
  */
 struct can_filter {
-    canid_t can_id;    /**< CAN ID */
-    canid_t can_mask;  /**< Mask */
+#if defined(MCU_SAMD5X)
+    can_filter_conf_t can_filter_conf;      /**< CAN filter configuration for same54 */
+    can_filte_type_t can_filter_type;       /**< CAN filtertype for same54 */
+#endif
+    canid_t can_id;    /**< CAN ID, for same54 CAN ID 1*/
+    canid_t can_mask;  /**< Mask, for same54 CAN ID 2 */
```

Comment:
So this field has a different meaning on SAM E5x?

Reply:
It depends from the filter configuration applied (in the code `can_filte_type_t`). In case of **CAN_FILTER_TYPE_CLASSIC** it will be the same (can_id, can_mask which have an AND relation) but for other configurations it's not the case. ID1 and ID2 is the naming given in the datasheet

Reply:
So SAM E5x supports more filter modes than the classic one?
This needs some better documentation then. Are those standard CAN features?

Reply:
I compared different CPUs and their CAN controllers provide different features concerning the CAN filters (STM provide mask-identifier feature and dual filter feature, ATMEL T89C51CC01 provides only mask-identifier, MCP2515 as a standalone controller provides only mask-identifier feature). So the standard and always present feature is the mask-identifier (single bit comparison) feature I would say. What do you think about using a unique filter structure for the SAME5x family?